mesonbuild/backend/ninjabackend.py

```diff
@@ -2317,7 +2323,8 @@ def generate_cython_compile_rules(self, compiler: 'Compiler') -> None:
 
     def generate_rust_compile_rules(self, compiler):
         rule = self.compiler_to_rule_name(compiler)
-        command = compiler.get_exelist() + ['$ARGS', '$in']
+        command = compiler.get_exelist() + ['--notify', '$notifyfile', '--emit', 'metadata', '--json=artifacts', '--error-format=json', '$ARGS', '$in']
```

Comment:
This seems like something we should generate two of, one to build rlibs with a wrapper and the other to execute rustc directly.

Reply:
That's a temporary hack to experiment, ideally rustc should support --notify so we don't need a wrapper and don't need to switch to json output.